### PR TITLE
feat: Add goal tracking to the export feature, closes #302

### LINKED ISSUE #302: Add goal tracking to the export feature
When the user clicks the export button it should be tracked.

1. Add GATSBY_FATHOM_EXPORT to your .env.development (any value will due)
2. Export FATHOM_EXPORT from `./constants`
2. Import the `trackGoal` function from the `tracking` feature. 
2. Import `FATHOM_EXPORT` from `./constants`
3. When button is clicked call `trackGoal` with `FATHOM_EXPORT` as param
4. Test and make sure the console logs out `Track fathom goal <FATHOM_EXPORT>)`

I will add a real value to Netlify env variables so this will be tracked properly by fathom in production.

### PR DESCRIPTION
**Pull request by olavea**

### CODE REVIEW

**Comment by olavea** [comment]
6. console.log("Track goal", {FATHOM_EXPORT}) #302
05a7dfe

365DaysOfPOWprogramming-day-151_{FATHOM_EXPORT}_302-357

**Comment by olavea** [comment]
- [ ] When the user clicks the export button it should be tracked.
- [ ] 
- [ ] Add GATSBY_FATHOM_EXPORT to your .env.development (any value will due)
- [ ] Export FATHOM_EXPORT from ./constants
- [ ] Import the trackGoal function from the tracking feature.
- [ ] Import FATHOM_EXPORT from ./constants
- [ ] When button is clicked call trackGoal with FATHOM_EXPORT as param
- [ ] Test and make sure the console logs out Track fathom goal <FATHOM_EXPORT>)
- [ ] I will add a real value to Netlify env variables so this will be tracked properly by fathom in production.

**Comment by olavea** [comment]
Why can I not log in with my batcatdad user (= 🦇 + 😺  + 🔨 )  in this deploy preview?
<details><summary>Browser metadata</summary>





```
Path:      /login/
Browser:   Chrome 90.0.4430.212 on Mac OS 10.15.6
Viewport:  1511 x 1081 @1x
Language:  en-US
Cookies:   Enabled
```

[View on BrowserStack](https://live.browserstack.com/dashboard#url=https%3A%2F%2Fdeploy-preview-305--pow-app.netlify.app%2Flogin%2F&os=OS+X&os_version=Catalina&browser=Chrome&browser_version=90.0&resolution=1511x1081&speed=1&start=true)



</details>

[Open Deploy Preview](https://deploy-preview-305--pow-app.netlify.app/login/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=60b4a331ac2d760008fb6014&commentID=60b4a86fa84d5e1d4727d2b4&resolution=resolved)

**Comment by raae** [comment]
> **A reviewer left a comment:**
> 
> Why can I not log in with my batcatdad user in this deploy preview?
> 
> Browser metadata
> [Open Deploy Preview](https://deploy-preview-305--pow-app.netlify.app/login/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=60b4a331ac2d760008fb6014&commentID=60b4a7496a1bd9ed5a2f6375&resolution=resolved)

Deploy preview is connected to a seperate test database.

File: src/features/profile/HouseKeepingCard.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ import { CardContentSection } from "../../components"
 import { Link } from "../navigation"
 import { selectAllEntries } from "../entries"
 
+import {FATHOM_EXPORT} from "../../../src/constants"
+import { trackGoal } from "../../../src/features/tracking/index";
```
Comment (raae) [inline src/features/profile/HouseKeepingCard.js:20]: **suggestion:** No need for the "/index" at the end.

**suggestion:** As little `../` as possible. `src` should never be a part of the import path. I think `../tracking` and `../../constants` would be correct

File: src/features/profile/HouseKeepingCard.js [hunk 2]
```diff
@@ -27,13 +30,23 @@ const HouseKeepingCard = () => {
     exportAllEntriesText = `No entries to export`
   }
 
+  // let FATHOM_EXPORT = "Pieces_Of_Eight"
```
Comment (raae) [inline src/features/profile/HouseKeepingCard.js:33]: **nitpick:** Remove unused comments

File: src/features/profile/HouseKeepingCard.js [hunk 3]
```diff
@@ -27,13 +30,23 @@ const HouseKeepingCard = () => {
...
+
+  // const trackGoal = () => {};
+
   const exportAllEntries = async (event) => {
     setIsPending(true)
     event.preventDefault()
 
+    // The intent of the user is to export her fistful_of_Pieces_Of_Eight
+    trackGoal(FATHOM_EXPORT)
+    console.log("Track fathom goal", {FATHOM_EXPORT})
```
Comment (raae) [inline src/features/profile/HouseKeepingCard.js:43]: **suggestion:** Remove as this is logged by the trackGoal function itself

**Review by raae — CHANGES_REQUESTED** [review]

File: src/constants.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ export const STRIPE_YEARLY_PLAN_ID = process.env.GATSBY_STRIPE_YEARLY_PLAN_ID
 export const SESSION_LENGTH =
   parseInt(process.env.GATSBY_USERBASE_SESSION_LENGTH) || 24 * 7 // a week
 
+export const FATHOM_EXPORT = process.env.GATSBY_FATHOM_EXPORT
```
[the remark this answers is not in the document]
Comment (olavea) [reply src/constants.js]: @olavea
suggestion: Move down to FATHOM_SUPPORT #302
be6a916

File: src/features/profile/HouseKeepingCard.js (hunk 1 above)
Comment (olavea) [reply src/features/profile/HouseKeepingCard.js]: ../tracking and ../../constants #302
0ca89a1

File: src/features/profile/HouseKeepingCard.js (hunk 2 above)
Comment (olavea) [reply src/features/profile/HouseKeepingCard.js]: nitpick: Remove unused comments …
be7edc7

File: src/features/profile/HouseKeepingCard.js (hunk 3 above)
Comment (olavea) [reply src/features/profile/HouseKeepingCard.js:38]: A. remove line 41 + 42 + 43 ? or
B. remove line 42  ? 

I go with B.
😸

Comment (olavea) [reply src/features/profile/HouseKeepingCard.js]: suggestion: Remove console.log #302
3743d24

**Comment by olavea** [comment]
So where do I go to sign up for that separate test database?
> > Why can I not log in with my batcatdad user in this deploy preview?
> Deploy preview is connected to a separate test database.

**Comment by olavea** [comment]
Can I merge this now Queen @raae 👑 ?

**Review by raae — APPROVED** [review]

**Comment by raae** [comment]
> Can I merge this now Queen @raae 👑 ?

Yes you may.

**Comment by raae** [comment]
I have added the goal id to Netlify, so if you test the preview now you'll see it being logged in the console.

User that works: username: may21sth password: may21stmay21st

**Pull request merged**